spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/metadata/MongoDbMetadataStore.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.mongodb.metadata;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.integration.metadata.ConcurrentMetadataStore;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import com.mongodb.WriteResult;
+
+/**
+ * MongoDbMetadataStore implementation of {@link ConcurrentMetadataStore}.
+ * Use this {@link org.springframework.integration.metadata.MetadataStore} to
+ * achieve meta-data persistence shared across application instances and
+ * restarts.
+ *
+ * @author Senthil Arumugam, Samiraj Panneer Selvam
+ * @since 4.2
+ *
+ */
+public class MongoDbMetadataStore implements ConcurrentMetadataStore {
+
+	protected final Log logger = LogFactory.getLog(getClass());
+	private final MongoTemplate template;
+	private final String DEFAULT_COLLECTION_NAME = "metadatastore";
+	private final String collectionName;
+	private final String ID_FIELD = "_id";
+	private final String VALUE = "value";
+	
+	/**
+	 * 
+	 * Configures the MongoDbMetadataStore by provided {@link MongoTemplate} and
+	 * default collection name - {@link #DEFAULT_COLLECTION_NAME}.
+	 * 
+	 * @param template The mongodb template
+	 * @param collectionName the collection name where it persists the data
+	 */
+	public MongoDbMetadataStore(MongoTemplate template, String collectionName) {
+		this.template = template;
+		this.collectionName = (StringUtils.hasText(collectionName)) ? collectionName
+				: DEFAULT_COLLECTION_NAME;
+	}
+
+	/**
+	 * Configures the MongoDbMetadataStore by provided {@link MongoTemplate} and
```

Comment:
The method JavaDocs must be `imperative`, In other words `Configure` not  `Configures`